src/responder/gasQueue.ts

```diff
@@ -94,7 +167,32 @@ export class GasQueueItem {
     }
 }
 
+interface GasQueueSerialisation {
+    readonly queueItems: Array<GasQueueItemSerialisation>;
+    readonly emptyNonce: number;
+    readonly replacementRate: number;
+    readonly maxQueueDepth: number;
+}
+
 export class GasQueue {
+    public static deserialise(serialisation: GasQueueSerialisation): GasQueue {
+        return new GasQueue(
+            serialisation.queueItems.map(i => GasQueueItem.deserialise(i)),
+            serialisation.emptyNonce,
+            serialisation.replacementRate,
+            serialisation.maxQueueDepth
+        );
+    }
+
+    public static serialise(queue: GasQueue): GasQueueSerialisation {
+        return {
+            queueItems: queue.queueItems.map(i => GasQueueItem.serialise(i)),
+            emptyNonce: queue.emptyNonce,
+            maxQueueDepth: queue.maxQueueDepth,
+            replacementRate: queue.replacementRate
+        };
+    }
+
```

Comment:
I'm not sure I like this approach of having a `-Serialisation` interface for each class that we want to serialise. While it adds some type safety, it does not enforce that all the serializable objects have the same "serialise"/"deserialise" methods; and the type safety is in a place where is probably not really useful (after you serialize an object, you are usually about to write it to DB after all).

I was thinking of a simple common interface like this:

```
interface Serialisable {
    serialise(): object;
}
```

the serialise() object is then called on the item itself rather than being a static class method.
A downside is that static methods cannot be specified in the interface, so still there is no way of forcing subclasses to implement a `deserialise` method.

Another concern is that the deserialisation is currently not checking for any kind of error (e.g. missing or wrong fields, or unexpected fields); but perhaps this should be a separate ticket.

Reply:
Yeah, `deserialise` is the problem. Since I cant have an interface for `deserialise` I'd rather keep both methods together.

Regarding errors, this deserialise isnt a checking method - we should never use it for data that isnt under our control. We should always be sure of the soundness of the data before calling deserialise - in this case soundness is ensured since we know the data was serialised by the `serialise` method. Later on we may need to add some specific checks (maybe when we make backward breaking changes to the serialise method)

Reply:
Yeah, my concern about error checking was to protect from
- database corruptions
- bugs in the (de)serialisation code, at least to catch them
- version changes as you mentioned.

Anyway, I agree that it's not the most pressing issue.